### PR TITLE
Add support for vectorization

### PR DESCRIPTION
**Pull request by aacostadiaz**
This PR add changes in the tree definition to add support for vectorisation in level one operators.

### CODE REVIEW

File: src/sb_handle/kernel_constructor.hpp [hunk 1]
```diff
@@ -244,8 +260,9 @@ static SYCL_BLAS_INLINE cl::sycl::event execute_tree(queue_t q_,
           cl::sycl::range<1>{globalSize}, cl::sycl::range<1>{localSize}};
```
[the remark this answers is not in the document]
Comment (OuadiElfarouki) [reply src/sb_handle/kernel_constructor.hpp:260]: Oh never-mind I see it's handled by `get_size_vector<vector_size>()` !

File: src/interface/blas1/backend/nvidia_gpu.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename nvidia_gpu.hpp
+ *
+ */
+
+#ifndef SYCL_BLAS_BLAS1_NVIDIA_GPU_HPP
+#define SYCL_BLAS_BLAS1_NVIDIA_GPU_HPP
+
+class HardwareSpec {
```
Comment (OuadiElfarouki) [inline src/interface/blas1/backend/nvidia_gpu.hpp:28]: `blas` & `backend` namespaces needed here ;)

File: src/sb_handle/kernel_constructor.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (aacostadiaz) [reply src/sb_handle/kernel_constructor.hpp:260]: Yes, I fix this after you comment! Good catch!

File: src/interface/blas1/backend/nvidia_gpu.hpp (hunk 2 above)
Comment (aacostadiaz) [reply src/interface/blas1/backend/nvidia_gpu.hpp:28]: done

**Comment by muhammad-tanvir-1211** [comment]
Closing this PR as the performance of vectorization doesn't look promising.

**Pull request closed without merging**